### ISSUE TITLE
Accessibility fixes for Plone Classic

### ISSUE THREAD

**Issue by thomasmassmann** [plone/Products.CMFPlone#3383]
## Abstract

Increase the accessibility in Plone Classic for Editors.

## Motivation

Plone Classic has several minor and major accessibility issues for editors. Editors are currently not able to work with Plone when assistive technologies are required.

Some issues exist because of too generic wording for labels and descriptions for some fields. Some issues exist because of the internal structure for widgets (a form hint/description should always be between the label and the widget). Some widgets are not accessible at all when assistive technologies are required (e.g. `SelectFieldWidget`, `AjaxSelectFieldWidget` — both because the underlaying `select2` and it’s implementation in Plone is not accessible at all) — editors currently hear “edit text” for those fields (hint: the label references a now hidden input field, and the new input field created by `select2` has no meaningful label; also the selected option(s) are not available to screen readers).

Widgets containing one or more checkboxes or radio buttons are not accessible for users requiring assistive technologies because of a missing group context. This group context should be a fieldset with a proper legend. But fieldsets are used to create groups (and tabs using `pat-autotoc`), and nested fieldsets are not recommended because of several issues with screen readers.

## Assumptions

Plone Classic would benefit from a more accessible user interface for editors. Because the implementation requires changes in multiple Plone core packages it makes sense to PLIP this for Plone core. Individual PRs will be created in the affected packages.

## Proposal & Implementation

TODO: show detailed a11y errors and possible solutions, link to the affected package

## Deliverables

Update form and widget related code in:
- plone.app.z3cform
- plone.z3cform

Update theme for the changed HTML structure in the packages mentioned above:
- plonetheme.barceloneta

## Risks

- Custom themes might have to be re-evaluated because of template/css changes.
- Add-ons might have to be updated to be compatible with the changes in the form templates

## Participants

- Thomas Massmann
- Veit Schiele

**Comment by ale-rt** [comment]
CC @plone/framework-team

**Comment by thomasmassmann** [comment]
> CC @plone/framework-team

Hi @ale-rt ,

I didn’t mention the framework team yet because it’s not finished. From the docs:

> Mention the @plone/framework-team when the PLIP is information complete!

😃

**Comment by jensens** [comment]
Would like to see this finished. I put it on the 6 project board anyway to not get lost.

**Comment by thomasmassmann** [comment]
I need to find the time to record and document the current issues for this PLIP in detail (e.g. video using a screen reader).

**Comment by MrTango** [comment]
> > CC @plone/framework-team
> 
> Hi @ale-rt ,
> 
> I didn’t mention the framework team yet because it’s not finished. From the docs:
> 
> > Mention the @plone/framework-team when the PLIP is information complete!
> 
> smiley

wasn't mentioned and was not a draft ;)

**Comment by tisto** [comment]
I move this to stalled. No activity since 2022.

**Comment by tisto** [comment]
@plone/classicui-team if you want to keep this PLIP, please assign it to a milestone (Plone 6.2, Plone 7, Future). Otherwise I'd suggest closting this PLIP. There has been no activity since 2022.
